src/ansys/fluent/core/fluent_connection.py

```diff
@@ -436,7 +436,13 @@ def __init__(
         # throws, we should not proceed.
         # TODO: Show user-friendly error message.
         if pyfluent.CHECK_HEALTH:
-            self.health_check.check_health()
+            connected = timeout_loop(
```

Comment:
Instead of adding a loop here, we can add a timeout directly in the grpc method call: https://github.com/grpc/grpc/blob/master/examples/python/timeout/greeter_client.py

Also, please make the timeout variable configurable via module-level global var.

Reply:
In the defect case, does the `self.health_check.check_health()` return `NOT_SERVING`? In that case, we should simply throw from here.

Reply:
There are 4 possibilities UNKNOWN, SERVING, NOT_SERVING and SERVICE_UNKNOWN therefore we are checking explicitly for SERVING only.

Reply:
> Instead of repeatedly calling the method within a loop here, we can add the timeout directly in the grpc method call: https://github.com/grpc/grpc/blob/master/examples/python/timeout/greeter_client.py
> 
> Also, please make the timeout variable configurable via module-level global var.

@mkundu1 We have `wait_for_server` method on HealthCheckService where we are already checking whether grpc.StatusCode.DEADLINE_EXCEEDED or not. We are not using `wait_for_server` anywhere. We can use it here right ?

Reply:
> Instead of repeatedly calling the method within a loop here, we can add the timeout directly in the grpc method call: https://github.com/grpc/grpc/blob/master/examples/python/timeout/greeter_client.py
> 
> Also, please make the timeout variable configurable via module-level global var.

Did the original solution repeatedly poll the server? That would allow the code to keep the user informed about what's cooking between polls (particularly useful for durations close to the timeout).
@mkundu1 @hpohekar

Reply:
When we reach at this code, the Fluent server should already be running (for `launch_fluent` we have already parsed the generated server-info file and for `connect_to_fluent` the server-info file from the server has been provided). We don't need any polling here. This check_health is a single call to check the sanity of the already running server. If there are any error from the running server, this check_health will fail instead of a later gRPC call.
@seanpearsonuk @hpohekar

Reply:
The server can be already running but cortex can still execute more code in the meantime. I think that this is the issue. If cortex gets into a bad state then the remote health check call can hang.